oap-server/server-storage-plugin/storage-banyandb-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/banyandb/BanyanDBSchema.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.storage.plugin.banyandb;
+
+import com.google.common.io.CharStreams;
+import com.google.protobuf.TextFormat;
+import org.apache.skywalking.banyandb.Database;
+import org.apache.skywalking.banyandb.Query;
+import org.apache.skywalking.banyandb.Schema;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class BanyanDBSchema {
+    private final Schema.TraceSeries traceSeries;
+    private final Set<String> fields = new LinkedHashSet<>();
+    private final Map<String, Integer> fieldsIndexMapping = new HashMap<>();
+
+    private int traceStateIndex;
+    private int traceIdIndex;
+    private Schema.FieldSpec.FieldType traceStateType;
+    private int valIntError;
+    private String valStringError;
```

Comment:
I am not sure how you organize the codes, what are the principle, but why this `BanyanDBSchema` directly have so many trace fields? What does it represent? If this is a schema, trace model should be a sub-class, isn't it?

Reply:
Since `traceState` and `traceId` are defined as reserved fields, in particular the `traceState`, it could be either String or Int. I have to parse it while the plugin is starting.

Reply:
Let's talk tomorrow. I am pretty sure this is not necessary.

Reply:
The whole schema is BanyanDB's internal thing. If there is version type convert, it is builder's responsibility. https://github.com/apache/skywalking/pull/7431#discussion_r686052303 is the key.